### ISSUE TITLE
Fix flaky tests

### ISSUE THREAD

**Issue by mtrezza** [parse-community/parse-server#7180]
### New Issue Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).
- [x] I can reproduce the issue with the [latest version of Parse Server](https://github.com/parse-community/parse-server/releases). 

### Issue Description
Flaky tests impede the CI process. With the increasing number of test scenarios to ensure Parse Server compatibility with recent versions of MongoDB, Postgres and Node.js, the likeliness of flaky tests disrupting the CI process also increases. In addition, GitHub Actions currently does not support to re-run individual jobs which makes the manual re-run a tedious effort.

### Expected outcome
Identify root cause of flakiness and fix for tests to be stable.

The following lists flaky tests to analyze and possibly find a pattern to determine the root cause. The list should also be a helper for maintainers to identify flaky tests when reviewing PRs. Feel free to update the list if you encounter a flaky test that is not in the list yet.

____

*Last list update: 2021-02-24*

### Flaky tests

_Tests which still have to be analyzed to fix the flakiness._

<details>
<summary>Schema Performance test schema update class</summary>

```
Error: Could not add field fooOne
```
</details>

<details>
<summary>Installations update ios fails with new deviceToken and no installationId</summary>

```
  Message:
    Failed
  Stack:
    Error: Failed
        at <Jasmine>
        at UserContext.<anonymous> (/home/runner/work/parse-server/parse-server/spec/AuthenticationAdapters.spec.js:1671:7)
        at <Jasmine>
        at processTicksAndRejections (internal/process/task_queues.js:97:5)
```
</details>

<details>
<summary>Parse.User testing only creates a single session for an installation / user pair (#2885)</summary>

```
 Message:
  Expected 2 to be 1.
Stack:
  Error: Expected 2 to be 1.
      at <Jasmine>
      at query.find.then.results (/home/runner/work/parse-server/parse-server/spec/ParseUser.spec.js:1294:32)
      at process._tickCallback (internal/process/next_tick.js:68:7)
Message:
  Expected 'r:a49facd19f31028da42c3058ce5c35a3' to be 'r:0bcc75ade014c3ecbc500a5a29606929'.
Stack:
  Error: Expected 'r:a49facd19f31028da42c3058ce5c35a3' to be 'r:0bcc75ade014c3ecbc500a5a29606929'.
      at <Jasmine>
      at query.find.then.results (/home/runner/work/parse-server/parse-server/spec/ParseUser.spec.js:1295:48)
      at process._tickCallback (internal/process/next_tick.js:68:7)
```
- Observed in environment(s): Postgres 12, Postgis 3.0
</details>

<details>
<summary>ParseLiveQuery expect afterEvent delete</summary>

```
 Message:
  Failed
Stack:
  Error: Failed
      at <Jasmine>
      at UserContext.it (/home/runner/work/parse-server/parse-server/spec/AuthenticationAdapters.spec.js:1671:7)
      at process._tickCallback (internal/process/next_tick.js:68:7)
```
- Observed in environment(s): Node 10
</details>

<details>
<summary>Apple Game Center Auth adapter validateAuthData should validate</summary>

```
Message:
  Error: Timeout - Async function did not complete within 20000ms (set by jasmine.DEFAULT_TIMEOUT_INTERVAL)
Stack:
  Error: Timeout - Async function did not complete within 20000ms (set by jasmine.DEFAULT_TIMEOUT_INTERVAL)
      at <Jasmine>
      at ontimeout (timers.js:436:11)
      at tryOnTimeout (timers.js:300:5)
      at listOnTimeout (timers.js:263:5)
      at Timer.processTimers (timers.js:223:10)
```
- Observed in environment(s): Node 10
</details>

<details>
<summary>Cloud Code test beforeSave with invalid field</summary>

```
  Message:
    Failed: deadlock detected
  Stack:
    error properties: Object({ length: 317, severity: 'ERROR', code: '40P01', detail: 'Process 151 waits for AccessExclusiveLock on relation 50837 of database 19568; blocked by process 156.
    Process 156 waits for AccessShareLock on relation 50712 of database 19568; blocked by process 151.', hint: 'See server log for query details.', position: undefined, internalPosition: undefined, internalQuery: undefined, where: undefined, schema: undefined, table: undefined, dataType: undefined, constraint: undefined, file: 'deadlock.c', routine: 'DeadLockReport' })
        at <Jasmine>
        at Parser.parseErrorMessage (/home/runner/work/parse-server/parse-server/node_modules/pg-protocol/dist/parser.js:278:15)
        at Parser.handlePacket (/home/runner/work/parse-server/parse-server/node_modules/pg-protocol/dist/parser.js:126:29)
        at Parser.parse (/home/runner/work/parse-server/parse-server/node_modules/pg-protocol/dist/parser.js:39:38)
        at Socket.stream.on (/home/runner/work/parse-server/parse-server/node_modules/pg-protocol/dist/index.js:10:42)
        at Socket.emit (events.js:198:13)
        at Socket.EventEmitter.emit (domain.js:448:20)
        at addChunk (_stream_readable.js:288:12)
        at readableAddChunk (_stream_readable.js:269:11)
        at Socket.Readable.push (_stream_readable.js:224:10)
        at TCP.onStreamRead [as onread] (internal/stream_base_commons.js:94:17)
```
</details>

<details>
 <summary>rest create test specified session length</summary>

```
Message:
  Expected 28 to equal 27.
Stack:
  Error: Expected 28 to equal 27.
  at rest.create.then.then.r (/home/runner/work/parse-server/parse-server/spec/rest.spec.js:602:37)
  at process._tickCallback (internal/process/next_tick.js:68:7)
```
- Observed in environment(s): ?
</details>

<details>
<summary>ParseGraphQLServer Auto API Schema Objects Queries Get read preferences should support includeReadPreference argument</summary>

```
Error: GraphQL error: Object not found.
```
</details>

<details>
 <summary>ParseGraphQLServer Auto API Schema Objects Queries Find read preferences should support readPreference argument</summary>

```
Message:
  Expected false to be true.
Stack:
  Error: Expected false to be true.
  at UserContext. (/home/runner/work/parse-server/parse-server/spec/ParseGraphQLServer.spec.js:5512:52)
  at processTicksAndRejections (internal/process/task_queues.js:93:5)
```
- Observed in environment(s): ?
</details>

<details>
<summary>ParseGraphQLServer Auto API Schema Files Mutations Create should return File object</summary>

```
Message:
  TypeError: Cannot read property 'clientMutationId' of null
Stack:
  at UserContext. (/home/runner/work/parse-server/parse-server/spec/ParseGraphQLServer.spec.js:6603:43)
  at processTicksAndRejections (internal/process/task_queues.js:97:5)
```
</details>

### Possibly fixed

_Tests for which a PR has been merged to fix the flakiness. These are currently under observation to confirm the fix. If you observe such a test to fail, please post a comment for further evaluation._

by #7187:

<details>
 <summary>ParseServerRESTController transactions should handle a batch request with transaction = true</summary>

```
Unhandled promise rejection: [object Object],[object Object]
```
- Observed in environment(s):
  - Mongo 4.2, ReplicaSet, WiredTiger, Node 10
  - Mongo 4.0, ReplicaSet, WiredTiger, Node 14
</details>

by #7196:

<details>
 <summary>Cloud Code cloud jobs should define a job</summary>

```
Unhandled promise rejection: ParseError: 101 Object not found.
```
- Observed in environment(s):
  - Mongo 4.4, ReplicaSet, WiredTiger, Node 14
</details>

<details>
 <summary>Cloud Code cloud jobs should define a job</summary>

```
Message:
  Unhandled promise rejection: error: relation "_JobStatus" does not exist (line 1180)
Stack:
  error properties: Object({ length: 108, severity: 'ERROR', code: '42P01', detail: undefined, hint: undefined, position: '8', internalPosition: undefined, internalQuery: undefined, where: undefined, schema: undefined, table: undefined, dataType: undefined, constraint: undefined, file: 'parse_relation.c', routine: 'parserOpenTable' })
  at <Jasmine>
  at Parser.parseErrorMessage (/home/runner/work/parse-server/parse-server/node_modules/pg-protocol/dist/parser.js:278:15)
  at Parser.handlePacket (/home/runner/work/parse-server/parse-server/node_modules/pg-protocol/dist/parser.js:126:29)
  at Parser.parse (/home/runner/work/parse-server/parse-server/node_modules/pg-protocol/dist/parser.js:39:38)
  at Socket.stream.on (/home/runner/work/parse-server/parse-server/node_modules/pg-protocol/dist/index.js:10:42)
  at Socket.emit (events.js:198:13)
  at Socket.EventEmitter.emit (domain.js:448:20)
  at addChunk (_stream_readable.js:288:12)
  at readableAddChunk (_stream_readable.js:269:11)
  at Socket.Readable.push (_stream_readable.js:224:10)
  at TCP.onStreamRead [as onread] (internal/stream_base_commons.js:94:17)
```
- Observed in environment(s):
  - Postgres 11, Postgis 3.1, Node 14.15.5
</details>

<details>
 <summary>Cloud Code cloud jobs should run with master key basic auth</summary>

```
Message:
  Unhandled promise rejection: ParseError: 101 Object not found.
Stack:
  error properties: Object({ code: 101 })
  Error: Object not found.
      at /home/runner/work/parse-server/parse-server/lib/Controllers/DatabaseController.js:49:3084
      at <Jasmine>
      at processTicksAndRejections (internal/process/task_queues.js:93:5)
```
- Observed in environment(s):
  - Redis Cache, Node 14.15.5
</details>

by #7212:
<details>
<summary>RedisCacheAdapter should find un-expired records</summary>

```
Message:
  Expected null to equal 'world'.
Stack:
  Error: Expected null to equal 'world'.
  at /home/runner/work/parse-server/parse-server/spec/RedisCacheAdapter.spec.js:97:36
  at processTicksAndRejections (internal/process/task_queues.js:93:5)
Message:
  Expected null not to equal null.
Stack:
  Error: Expected null not to equal null.
  at /home/runner/work/parse-server/parse-server/spec/RedisCacheAdapter.spec.js💯40
  at processTicksAndRejections (internal/process/task_queues.js:93:5)
```
</details>

bv #7213:
<details>
 <summary>PushController properly creates _PushStatus</summary>

```
Message:
  Expected 'pending' to be 'running'.
Stack:
  Error: Expected 'pending' to be 'running'.
  at reconfigureServer.then.then.then.then.then.catch.then (/home/runner/work/parse-server/parse-server/spec/PushController.spec.js:618:48)
at process._tickCallback (internal/process/next_tick.js:68:7)
```
</details>

by #7228:

<details>
<summary>PostgresStorageAdapter should use index for caseInsensitive query using default indexname</summary>

```
  Message:
    error: duplicate key value violates unique constraint "_User_pkey" (line 434)
  Stack:
    error properties: Object({ length: 219, severity: 'ERROR', code: '23505', detail: 'Key ("objectId")=(d3c9d0cbebeeb2efcc62a95f09255ff2) already exists.', hint: undefined, position: undefined, internalPosition: undefined, internalQuery: undefined, where: undefined, schema: 'public', table: '_User', dataType: undefined, constraint: '_User_pkey', file: 'nbtinsert.c', routine: '_bt_check_unique' })
        at <Jasmine>
        at Parser.parseErrorMessage (/home/runner/work/parse-server/parse-server/node_modules/pg-protocol/dist/parser.js:278:15)
        at Parser.handlePacket (/home/runner/work/parse-server/parse-server/node_modules/pg-protocol/dist/parser.js:126:29)
        at Parser.parse (/home/runner/work/parse-server/parse-server/node_modules/pg-protocol/dist/parser.js:39:38)
        at Socket.stream.on (/home/runner/work/parse-server/parse-server/node_modules/pg-protocol/dist/index.js:10:42)
        at Socket.emit (events.js:198:13)
        at Socket.EventEmitter.emit (domain.js:448:20)
        at addChunk (_stream_readable.js:288:12)
        at readableAddChunk (_stream_readable.js:269:11)
        at Socket.Readable.push (_stream_readable.js:224:10)
        at TCP.onStreamRead [as onread] (internal/stream_base_commons.js:94:17)
```
</details>

by #7265:
<details>
<summary>Parse.GeoPoint testing creating geo point exception two fields</summary>

```
Message:
  Failed: deadlock detected
Stack:
  error properties: Object({ length: 319, severity: 'ERROR', code: '40P01', detail: 'Process 150 waits for AccessExclusiveLock on relation 139374 of database 19581; blocked by process 154.
  Process 154 waits for AccessShareLock on relation 139249 of database 19581; blocked by process 150.', hint: 'See server log for query details.', position: undefined, internalPosition: undefined, internalQuery: undefined, where: undefined, schema: undefined, table: undefined, dataType: undefined, constraint: undefined, file: 'deadlock.c', routine: 'DeadLockReport' })
      at <Jasmine>
      at Parser.parseErrorMessage (/home/runner/work/parse-server/parse-server/node_modules/pg-protocol/dist/parser.js:278:15)
      at Parser.handlePacket (/home/runner/work/parse-server/parse-server/node_modules/pg-protocol/dist/parser.js:126:29)
      at Parser.parse (/home/runner/work/parse-server/parse-server/node_modules/pg-protocol/dist/parser.js:39:38)
      at Socket.stream.on (/home/runner/work/parse-server/parse-server/node_modules/pg-protocol/dist/index.js:10:42)
      at Socket.emit (events.js:198:13)
      at Socket.EventEmitter.emit (domain.js:448:20)
      at addChunk (_stream_readable.js:288:12)
      at readableAddChunk (_stream_readable.js:269:11)
      at Socket.Readable.push (_stream_readable.js:224:10)
      at TCP.onStreamRead [as onread] (internal/stream_base_commons.js:94:17)
```
- Observed in environment(s): Postgres
</details>

by #7267:

<details>
<summary>Cloud Code cloud jobs should set the message / success on the job</summary>

```
Message:
  Expected 'running' to equal 'succeeded'.
Stack:
  Error: Expected 'running' to equal 'succeeded'.
      at <Jasmine>
      at promise.then.then.jobStatus (/home/runner/work/parse-server/parse-server/spec/CloudCode.spec.js:1666:45)
      at process._tickCallback (internal/process/next_tick.js:68:7)
```
</details>

<details>
<summary>PushController should not enqueue push when device token is not set</summary>

```
  Message:
    Expected 'running' to be 'succeeded'.
  Stack:
    Error: Expected 'running' to be 'succeeded'.
        at <Jasmine>
        at query.find.then.results (/home/runner/work/parse-server/parse-server/spec/PushController.spec.js:1128:44)
        at process._tickCallback (internal/process/next_tick.js:68:7)
```
</details>

<details>
<summary>Parse.Push does not get stuck with _PushStatus 'running' on 1 installation added</summary>

```
Message:
  Expected 'running' to equal 'succeeded'.
Stack:
  Error: Expected 'running' to equal 'succeeded'.
      at <Jasmine>
      at reconfigureServer.then.then.then.then.then.results (/home/runner/work/parse-server/parse-server/spec/Parse.Push.spec.js:388:38)
      at process._tickCallback (internal/process/next_tick.js:68:7)
Message:
  Expected 1 to equal undefined.
Stack:
  Error: Expected 1 to equal undefined.
      at <Jasmine>
      at reconfigureServer.then.then.then.then.then.results (/home/runner/work/parse-server/parse-server/spec/Parse.Push.spec.js:390:37)
      at process._tickCallback (internal/process/next_tick.js:68:7)
```
</details>

### Fixed tests

_Tests that have not been observed anymore for a while and are therefore considered fixed._

**Comment by mtrezza** [comment]
@dplewis Do you have any idea how we can investigate this? Maybe add NewRelic to it to get the call stack when a test fails?

**Comment by davimacedo** [comment]
> @dplewis Do you have any idea how we can investigate this? Maybe add NewRelic to it to get the call stack when a test fails?

The way I do is basically creating a loop around the test and making it to execute let's say 100 times. By this way you can see it failing and you can debug. I am tackling the second one now.

**Comment by davimacedo** [comment]
https://github.com/parse-community/parse-server/pull/7196 probably solves items 2 and 3

**Comment by davimacedo** [comment]
Few more flaky tests that I've observed:

1) rest create test specified session length
49
  Message:
50
    Expected 28 to equal 27.
51
  Stack:
52
    Error: Expected 28 to equal 27.
53
        at <Jasmine>
54
        at rest.create.then.then.r (/home/runner/work/parse-server/parse-server/spec/rest.spec.js:602:37)
55
        at process._tickCallback (internal/process/next_tick.js:68:7)



1) Parse.GeoPoint testing creating geo point exception two fields
49
  Message:
50
    Failed: deadlock detected
51
  Stack:
52
    error properties: Object({ length: 319, severity: 'ERROR', code: '40P01', detail: 'Process 158 waits for AccessExclusiveLock on relation 139308 of database 19596; blocked by process 152.
53
    Process 152 waits for AccessShareLock on relation 139183 of database 19596; blocked by process 158.', hint: 'See server log for query details.', position: undefined, internalPosition: undefined, internalQuery: undefined, where: undefined, schema: undefined, table: undefined, dataType: undefined, constraint: undefined, file: 'deadlock.c', routine: 'DeadLockReport' })
54
        at <Jasmine>
55
        at Parser.parseErrorMessage (/home/runner/work/parse-server/parse-server/node_modules/pg-protocol/dist/parser.js:278:15)
56
        at Parser.handlePacket (/home/runner/work/parse-server/parse-server/node_modules/pg-protocol/dist/parser.js:126:29)
57
        at Parser.parse (/home/runner/work/parse-server/parse-server/node_modules/pg-protocol/dist/parser.js:39:38)
58
        at Socket.stream.on (/home/runner/work/parse-server/parse-server/node_modules/pg-protocol/dist/index.js:10:42)
59
        at Socket.emit (events.js:198:13)
60
        at Socket.EventEmitter.emit (domain.js:448:20)
61
        at addChunk (_stream_readable.js:288:12)
62
        at readableAddChunk (_stream_readable.js:269:11)
63
        at Socket.Readable.push (_stream_readable.js:224:10)
64
        at TCP.onStreamRead [as onread] (internal/stream_base_commons.js:94:17)


1) ParseGraphQLServer Auto API Schema Objects Queries Find read preferences should support readPreference argument
22
  Message:
23
    Expected false to be true.
24
  Stack:
25
    Error: Expected false to be true.
26
        at <Jasmine>
27
        at UserContext.<anonymous> (/home/runner/work/parse-server/parse-server/spec/ParseGraphQLServer.spec.js:5512:52)
28
        at <Jasmine>
29
        at processTicksAndRejections (internal/process/task_queues.js:93:5)

**Comment by mtrezza** [comment]
I added then to the list above. I'll mark tests as solved by https://github.com/parse-community/parse-server/pull/7196 if they don't occur anymore after some time.

**Comment by davimacedo** [comment]
"ParseServerRESTController transactions should handle a batch request with transaction = true" was actually also solved by https://github.com/parse-community/parse-server/pull/7187 so let's observe this one as well.

**Comment by mtrezza** [comment]
Thanks, updated.

**Comment by davimacedo** [comment]
One more:
1) PushController properly creates _PushStatus
  Message:
    Expected 'pending' to be 'running'.
  Stack:
    Error: Expected 'pending' to be 'running'.
        at <Jasmine>
        at reconfigureServer.then.then.then.then.then.catch.then (/home/runner/work/parse-server/parse-server/spec/PushController.spec.js:618:48)
        at process._tickCallback (internal/process/next_tick.js:68:7)

**Comment by mtrezza** [comment]
Added!

**Comment by davimacedo** [comment]
For "Parse.GeoPoint testing creating geo point exception two fields" I've run all geo point related tests 300 times for Postgres and Mongo in a row with no error. I remember in that time there was other strange messages happening for the Postgres database, so I guess the Postgres was in an inconsistent state maybe because of anther test but it does not look to have a problem with this test itself. I suggest we observe it more and investigate it again in the case it happens to fail.

**Comment by davimacedo** [comment]
This one is for Redis cache:
1) RedisCacheAdapter should find un-expired records
  Message:
    Expected null to equal 'world'.
  Stack:
    Error: Expected null to equal 'world'.
        at <Jasmine>
        at /home/runner/work/parse-server/parse-server/spec/RedisCacheAdapter.spec.js:97:36
        at <Jasmine>
        at processTicksAndRejections (internal/process/task_queues.js:93:5)
  Message:
    Expected null not to equal null.
  Stack:
    Error: Expected null not to equal null.
        at <Jasmine>
        at /home/runner/work/parse-server/parse-server/spec/RedisCacheAdapter.spec.js:100:40
        at <Jasmine>
        at processTicksAndRejections (internal/process/task_queues.js:93:5)

**Comment by dplewis** [comment]
I think I may have a fix for this one in another [branch](https://github.com/parse-community/parse-server/commit/d65de9ecddd6f48c1c44bca89fb0cb06be005807). 

```
1) Cloud Code cloud jobs should set the message / success on the job
  Message:
    Expected 'running' to equal 'succeeded'.
  Stack:
    Error: Expected 'running' to equal 'succeeded'.
        at <Jasmine>
        at promise.then.then.jobStatus (/home/runner/work/parse-server/parse-server/spec/CloudCode.spec.js:1666:45)
        at process._tickCallback (internal/process/next_tick.js:68:7)
Pending:
```

**Comment by davimacedo** [comment]
"rest create test specified session length" is now possibly fixed by https://github.com/parse-community/parse-server/pull/7208

**Comment by davimacedo** [comment]
> I think I may have a fix for this one in another [branch](https://github.com/parse-community/parse-server/commit/d65de9ecddd6f48c1c44bca89fb0cb06be005807). Or it fixes the
> `PushController properly creates _PushStatus` test
> 
> ```
> 1) Cloud Code cloud jobs should set the message / success on the job
>   Message:
>     Expected 'running' to equal 'succeeded'.
>   Stack:
>     Error: Expected 'running' to equal 'succeeded'.
>         at <Jasmine>
>         at promise.then.then.jobStatus (/home/runner/work/parse-server/parse-server/spec/CloudCode.spec.js:1666:45)
>         at process._tickCallback (internal/process/next_tick.js:68:7)
> Pending:
> ```

Cool! Could you please open a PR?

**Comment by davimacedo** [comment]
I've also couldn't reproduce "ParseGraphQLServer Auto API Schema Objects Queries Find read preferences should support readPreference argument" yet. I suggest now we observe more as most of the flaky tests were tackled. I will continue working in other stuff and come back to here when seeing something else failing.

**Comment by mtrezza** [comment]
Added the 2 above mentioned tests and one of them as possibly fixed by #7212.

**Comment by dplewis** [comment]
`PushController properly creates _PushStatus` possibly fixed by https://github.com/parse-community/parse-server/pull/7213
I added a description of the issue that you might find interesting

**Comment by mtrezza** [comment]
Updated list.

**Comment by mtrezza** [comment]
Added new flaky test: PostgresStorageAdapter should use index for caseInsensitive query using default indexname

**Comment by davimacedo** [comment]
One more: 1) ParseGraphQLServer Auto API Schema Files Mutations Create should return File object
22
  Message:
23
    TypeError: Cannot read property 'clientMutationId' of null
24
  Stack:
25
        at <Jasmine>
26
        at UserContext.<anonymous> (/home/runner/work/parse-server/parse-server/spec/ParseGraphQLServer.spec.js:6603:43)
27
        at <Jasmine>
28
        at processTicksAndRejections (internal/process/task_queues.js:97:5)

Reference: https://github.com/parse-community/parse-server/runs/1972047490?check_suite_focus=true

**Comment by mtrezza** [comment]
Added!

**Comment by cbaker6** [comment]
> Added new flaky test: PostgresStorageAdapter should use index for caseInsensitive query using default indexname

Should be fixed in #7228

**Comment by cbaker6** [comment]
Here: 

```
1) Parse.Push does not get stuck with _PushStatus 'running' on 1 installation added
49
  Message:
50
    Expected 'running' to equal 'succeeded'.
51
  Stack:
52
    Error: Expected 'running' to equal 'succeeded'.
53
        at <Jasmine>
54
        at reconfigureServer.then.then.then.then.then.results (/home/runner/work/parse-server/parse-server/spec/Parse.Push.spec.js:388:38)
55
        at process._tickCallback (internal/process/next_tick.js:68:7)
56
  Message:
57
    Expected 1 to equal undefined.
58
  Stack:
59
    Error: Expected 1 to equal undefined.
60
        at <Jasmine>
61
        at reconfigureServer.then.then.then.then.then.results (/home/runner/work/parse-server/parse-server/spec/Parse.Push.spec.js:390:37)
62
        at process._tickCallback (internal/process/next_tick.js:68:7)
```

**Comment by cbaker6** [comment]
Reference: https://github.com/parse-community/parse-server/pull/7228/checks?check_run_id=1983308358#step:8:48

```
Failures:
48
1) Parse.GeoPoint testing creating geo point exception two fields
49
  Message:
50
    Failed: deadlock detected
51
  Stack:
52
    error properties: Object({ length: 319, severity: 'ERROR', code: '40P01', detail: 'Process 149 waits for AccessExclusiveLock on relation 141165 of database 19581; blocked by process 152.
53
    Process 152 waits for AccessShareLock on relation 141040 of database 19581; blocked by process 149.', hint: 'See server log for query details.', position: undefined, internalPosition: undefined, internalQuery: undefined, where: undefined, schema: undefined, table: undefined, dataType: undefined, constraint: undefined, file: 'deadlock.c', routine: 'DeadLockReport' })
54
        at <Jasmine>
55
        at Parser.parseErrorMessage (/home/runner/work/parse-server/parse-server/node_modules/pg-protocol/dist/parser.js:278:15)
56
        at Parser.handlePacket (/home/runner/work/parse-server/parse-server/node_modules/pg-protocol/dist/parser.js:126:29)
57
        at Parser.parse (/home/runner/work/parse-server/parse-server/node_modules/pg-protocol/dist/parser.js:39:38)
58
        at Socket.stream.on (/home/runner/work/parse-server/parse-server/node_modules/pg-protocol/dist/index.js:10:42)
59
        at Socket.emit (events.js:198:13)
60
        at Socket.EventEmitter.emit (domain.js:448:20)
61
        at addChunk (_stream_readable.js:288:12)
62
        at readableAddChunk (_stream_readable.js:269:11)
63
        at Socket.Readable.push (_stream_readable.js:224:10)
64
        at TCP.onStreamRead [as onread] (internal/stream_base_commons.js:94:17)
```

**Comment by mtrezza** [comment]
Added `Parse.Push does not get stuck with _PushStatus 'running' on 1 installation added`; did not add `Parse.GeoPoint testing creating geo point exception two fields` as it is a duplicate

**Comment by mtrezza** [comment]
Added 2 new flaky tests

**Comment by cbaker6** [comment]
https://github.com/parse-community/parse-server/runs/2041751762#step:8:30

```
2) Installations update ios fails with new deviceToken and no installationId

  Message:

    Failed

  Stack:

    Error: Failed

        at <Jasmine>

        at UserContext.<anonymous> (/home/runner/work/parse-server/parse-server/spec/AuthenticationAdapters.spec.js:1671:7)

        at <Jasmine>

        at processTicksAndRejections (internal/process/task_queues.js:97:5)
```

**Comment by mtrezza** [comment]
Added!

**Comment by cbaker6** [comment]
https://github.com/parse-community/parse-server/runs/2048364609#step:8:48

```
1) Cloud Code test beforeSave with invalid field
  Message:
    Failed: deadlock detected
  Stack:
    error properties: Object({ length: 317, severity: 'ERROR', code: '40P01', detail: 'Process 151 waits for AccessExclusiveLock on relation 50837 of database 19568; blocked by process 156.
    Process 156 waits for AccessShareLock on relation 50712 of database 19568; blocked by process 151.', hint: 'See server log for query details.', position: undefined, internalPosition: undefined, internalQuery: undefined, where: undefined, schema: undefined, table: undefined, dataType: undefined, constraint: undefined, file: 'deadlock.c', routine: 'DeadLockReport' })
        at <Jasmine>
        at Parser.parseErrorMessage (/home/runner/work/parse-server/parse-server/node_modules/pg-protocol/dist/parser.js:278:15)
        at Parser.handlePacket (/home/runner/work/parse-server/parse-server/node_modules/pg-protocol/dist/parser.js:126:29)
        at Parser.parse (/home/runner/work/parse-server/parse-server/node_modules/pg-protocol/dist/parser.js:39:38)
        at Socket.stream.on (/home/runner/work/parse-server/parse-server/node_modules/pg-protocol/dist/index.js:10:42)
        at Socket.emit (events.js:198:13)
        at Socket.EventEmitter.emit (domain.js:448:20)
        at addChunk (_stream_readable.js:288:12)
        at readableAddChunk (_stream_readable.js:269:11)
        at Socket.Readable.push (_stream_readable.js:224:10)
        at TCP.onStreamRead [as onread] (internal/stream_base_commons.js:94:17)
```

**Comment by dplewis** [comment]
`Parse.GeoPoint testing creating geo point exception two fields` fixed by https://github.com/parse-community/parse-server/pull/7265

**Comment by mtrezza** [comment]
Updated, added 2, fixed 1

**Comment by dplewis** [comment]
Jobs and Push flaky tests are addressed by https://github.com/parse-community/parse-server/pull/7267
```
Cloud Code cloud jobs should set the message / success on the job
PushController should not enqueue push when device token is not set
Parse.Push does not get stuck with _PushStatus 'running' on 1 installation added
```

**Comment by mtrezza** [comment]
Nice, updated!

**Comment by mtrezza** [comment]
Added 

<details>
<summary>ParseGraphQLServer Auto API Schema Objects Queries Get read preferences should support includeReadPreference argument</summary>

```
Error: GraphQL error: Object not found.
```
</details>

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [6.1.0-alpha.8](https://github.com/parse-community/parse-server/releases/tag/6.1.0-alpha.8)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [6.1.0-beta.2](https://github.com/parse-community/parse-server/releases/tag/6.1.0-beta.2)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [6.1.0](https://github.com/parse-community/parse-server/releases/tag/6.1.0)

**Comment by dplewis** [comment]
Closing via https://github.com/parse-community/parse-server/pull/8468
